src/git-command-manager.ts

```diff
@@ -0,0 +1,399 @@
+import * as core from '@actions/core'
+import * as exec from '@actions/exec'
+import * as fshelper from './fs-helper';
+import * as io from '@actions/io';
+import * as path from 'path';
+
+export interface IGitCommandManager {
+    branchExists(remote: boolean, pattern: string): Promise<boolean>;
+    checkout(ref: string, startPoint: string): Promise<void>;
+    config(configKey: string, configValue: string): Promise<void>;
+    configExists(configKey: string): Promise<boolean>;
+    fetch(fetchDepth: number, refSpec: string[]): Promise<void>;
+    getWorkingDirectory(): string;
+    init(): Promise<void>;
+    lfsFetch(ref: string): Promise<void>;
+    lfsInstall(): Promise<void>;
+    log1(): Promise<void>;
+    remoteAdd(remoteName: string, remoteUrl: string): Promise<void>;
+    tagExists(pattern: string): Promise<boolean>;
+    tryClean(): Promise<boolean>;
+    tryConfigUnset(configKey: string): Promise<boolean>;
+    tryDisableAutomaticGarbageCollection(): Promise<boolean>;
+    tryGetFetchUrl(): Promise<string>;
+    tryReset(): Promise<boolean>;
+}
+
+export async function CreateCommandManager(
```

Comment:
we need to do some async stuff when initializing the GitCommandManager class, so this function is exposed and the ctor for GitCommandManager is private